packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -187,17 +183,37 @@ export class EpochTracker {
         fromCache: boolean = false,
     ) {
         if (error.errorType === OdspErrorType.epochVersionMismatch) {
-            const err = {
-                ...error,
-                fromCache,
-                clientEpoch: this.fluidEpoch,
-                serverEpoch: epochFromResponse ?? undefined,
-                fetchType,
-            };
-            this.logger.sendErrorEvent({ eventName: "EpochVersionMismatch" }, err);
-            assert(!!this.fileEntry, "File Entry should be set to clear the cached entries!!");
-            // If the epoch mismatches, then clear all entries for such file entry from cache.
-            await this.persistedCache.removeEntries(this.fileEntry);
+            try {
+                // This will only throw if it is an epoch error.
+                this.checkForEpochErrorCore(epochFromResponse, error.errorMessage);
+            } catch (epochError) {
+                const err = {
+                    ...epochError,
+                    fromCache,
+                    clientEpoch: this.fluidEpoch,
+                    serverEpoch: epochFromResponse ?? undefined,
+                    fetchType,
+                };
+                this.logger.sendErrorEvent({ eventName: "EpochVersionMismatch" }, err);
+                assert(!!this.fileEntry, "File Entry should be set to clear the cached entries!!");
+                // If the epoch mismatches, then clear all entries for such file entry from cache.
+                await this.persistedCache.removeEntries(this.fileEntry);
+                throw epochError;
+            }
+            // If it was categorised as epoch error but the epoch returned in response matches with the client epoch
+            // then it was coherency 409, so rethrow it as throttling error so that it can retried. Default throttling
+            // time is 1s.
+            this.logger.sendTelemetryEvent({ eventName: "Coherency409" }, error);
```

Comment:
I'd make it an error. Too many of these and the user experience is affected, so we want OCEs to see them quickly.

Reply:
Also we might be wrong in our logic and server might indeed point to epoch failure :) And then knowing about it sooner would help us fix the problem sooner